tools/lint-pr-commit-message.sh

```diff
@@ -25,21 +25,15 @@ if [ -z "${PR_ID}" ]; then
   echo " e.g. $0 <PR_NUMBER>"
   exit 1
 fi
-# Retrieve the first commit of the pull request via GitHub API
-# TODO: If we teach core-validate-commit to ignore "fixup!" and "squash!"
-#       commits and lint messages for all commits in the pull request
-#       we could simplify the following to:
-#         npx -q core-validate-commit --no-validate-metadata ${GH_API_URL}/repos/nodejs/node/pulls/${PR_ID}/commits
-if PR_COMMITS="$( curl -s ${GH_API_URL}/repos/nodejs/node/pulls/${PR_ID}/commits )"; then
-  if FIRST_COMMIT="$( node -p 'JSON.parse(process.argv[1])[0].url' "${PR_COMMITS}" 2> /dev/null )"; then
-    echo "Linting the first commit message for pull request ${PR_ID}"
-    echo "according to the guidelines at https://goo.gl/p2fr5Q."
-    # Print the commit message to make it more obvious what is being checked.
-    echo "Commit message for ${FIRST_COMMIT##*/} is:"
-    node -p 'JSON.parse(process.argv[1])[0].commit.message' "${PR_COMMITS}" 2> /dev/null
-    npx -q core-validate-commit --no-validate-metadata "${FIRST_COMMIT}"
-  else
-    echo "Unable to determine the first commit for pull request ${PR_ID}."
-    exit 1
-  fi
+
+PATCH=$( curl -sL https://github.com/nodejs/node/pull/${PR_ID}.patch | grep '^From \|^Subject: ' )
+if FIRST_COMMIT="$( echo "$PATCH" | awk '/^From [0-9a-f]{40} / { if (count++ == 0) print $2 }' )"; then
+  MESSAGE=$( echo "$PATCH" | awk '/^Subject: \[PATCH 1/ { gsub(/^Subject: \[PATCH[^\]]*\] /, ""); print }' )
```

Comment:
It looks like the `MESSAGE` logic isn't working -- Probably because the message has been filtered out of `PATCH` above?

e.g.
```
-bash-4.2$ bash tools/lint-pr-commit-message.sh 24574

*** Linting the first commit message for pull request 24574
*** according to the guidelines at https://goo.gl/p2fr5Q.
*** Commit message for 6f645d3675 is:


  ✔  6f645d367591a2d3734caa28dacdd7bcdcef3528
     ✔  1:7      Valid fixes url                           fixes-url
     ✔  0:0      blank line after title                    line-after-title
     ✔  0:0      line-lengths are valid                    line-length
     ✔  0:0      valid subsystems                          subsystem
     ✔  0:0      Title is formatted correctly.             title-format
     ✔  0:0      Title is <= 50 columns.                   title-length
-bash-4.2$
```

I would expect to see
```
...
*** Commit message for 6f645d3675 is:
tools: don't use GH API for commit message checks

Fixes: https://github.com/nodejs/node/issues/24567
...
```

The reason for printing the message is that it should make it obvious what is being linted (just in case, for example, the wrong SHA is used).

Reply:
```diff
diff --git a/tools/lint-pr-commit-message.sh b/tools/lint-pr-commit-message.sh
index 0bc873f..7ea75ab 100644
--- a/tools/lint-pr-commit-message.sh
+++ b/tools/lint-pr-commit-message.sh
@@ -28,12 +28,12 @@ fi

 PATCH=$( curl -sL https://github.com/nodejs/node/pull/${PR_ID}.patch | grep '^From \|^Subject: ' )
 if FIRST_COMMIT="$( echo "$PATCH" | awk '/^From [0-9a-f]{40} / { if (count++ == 0) print $2 }' )"; then
-  MESSAGE=$( echo "$PATCH" | awk '/^Subject: \[PATCH 1/ { gsub(/^Subject: \[PATCH[^\]]*\] /, ""); print }' )
+  MESSAGE=$( git show --quiet --format='format:%B' $FIRST_COMMIT )
   echo "
 *** Linting the first commit message for pull request ${PR_ID}
 *** according to the guidelines at https://goo.gl/p2fr5Q.
 *** Commit message for $(echo $FIRST_COMMIT | cut -c 1-10) is:
-      ${MESSAGE}
+${MESSAGE}
 "
   npx -q core-validate-commit --no-validate-metadata "${FIRST_COMMIT}"
 fi
```

e.g.
```
-bash-4.2$ bash tools/lint-pr-commit-message.sh 24574

*** Linting the first commit message for pull request 24574
*** according to the guidelines at https://goo.gl/p2fr5Q.
*** Commit message for 6f645d3675 is:
tools: don't use GH API for commit message checks

Fixes: https://github.com/nodejs/node/issues/24567

  ✔  6f645d367591a2d3734caa28dacdd7bcdcef3528
     ✔  1:7      Valid fixes url                           fixes-url
     ✔  0:0      blank line after title                    line-after-title
     ✔  0:0      line-lengths are valid                    line-length
     ✔  0:0      valid subsystems                          subsystem
     ✔  0:0      Title is formatted correctly.             title-format
     ✔  0:0      Title is <= 50 columns.                   title-length
-bash-4.2$
```